### PR TITLE
global: refactored `CFG` variables

### PR DESCRIPTION
**Pull request by dset0x**
* Replaces custom path concatenations with `os.path.join`

* Refactors commonly used paths to use config variables.

* ~~Renames `CFG_PREFIX` to `_CFG_PREFIX`.~~ updated by @jirikuncar

---

I would like some input on this in order to drop the WIP tag quicker:
Some references to `_CFG_PREFIX` remain and perhaps some `legacy/` files were better off remaining similar to their `master` equivalents.

If accepted, supersedes  #2818

### CODE REVIEW

File: invenio/legacy/ckeditor/connector.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def process_CKEditor_upload(form, uid, user_files_path, user_files_absolute_path
     @type user_files_path: string
     @param user_files_absolute_path: the base path on the server where
         the files should be saved.
-        Eg:C{%(CFG_PREFIX)s/var/data/comments/%(recid)s/%(uid)s}
+        Eg:C{%(_CFG_PREFIX)s/var/data/comments/%(recid)s/%(uid)s}
```
Comment (jirikuncar) [inline invenio/legacy/ckeditor/connector.py]: `%(_CFG_PREFIX)s/var/data` should have it's own config value

Comment (dset0x) [inline invenio/legacy/ckeditor/connector.py]: It already does, but this was not amended correctly - thanks!

File: invenio/base/config.py [hunk 2]
```diff
@@ -75,24 +75,29 @@
     'invenio.legacy.websubmit',
 ]
 
-CFG_PREFIX = distutils.sysconfig.get_config_var("prefix")
+_CFG_PREFIX = distutils.sysconfig.get_config_var("prefix")
```
Comment (jirikuncar) [inline invenio/base/config.py]: Please keep the variable name unchanged for now.

Comment (dset0x) [inline invenio/base/config.py]: @jirikuncar Perhaps attach a deprecation warning on access time?

Comment (jirikuncar) [inline invenio/base/config.py]: I'm not sure what is the best way to deprecate its usage, but there is an example:
```python
def cfg_prefix():
    from invenio.utils.deprecation import RemovedInInvenio22Warning
    warnings.warn('... CFG_PREFIX ...', RemovedInInvenio22Warning
    return distutils.sysconfig.get_config_var("prefix")

CFG_PREFIX = LocalProxy(cfg_prefix)
```

Comment (jirikuncar) [inline invenio/base/config.py]: We might also add list of deprecated config keys in `invenio/ext/confighacks.py`.

Comment (dset0x) [inline invenio/base/config.py]: Cool. I hadn't used `LocalProxy` before.

I will look into modifying `invenio/ext/logging/__init__.py:deprecated` to work with the new deprecation warning classes and then it should be a little more organized:

```python
@deprecated('Use the other variables instead', (2, 2))
@property
def cfg_prefix():
    return distutils.sysconfig.get_config_var("prefix")
CFG_PREFIX = LocalProxy(cfg_prefix)

Comment (jirikuncar) [inline invenio/base/config.py]: `invenio/ext/logging/__init__.py:deprecated` we could move it to `invenio/utils/deprecation.py`. WDYT?

Comment (dset0x) [inline invenio/base/config.py]: Seemed appropriate. Done.

I've also made the changes we discussed in #2818.

File: invenio/legacy/inveniocfg.py [hunk 3]
```diff
@@ -419,7 +419,7 @@ def cli_cmd_update_web_tests(conf):
     """
     print(">>> Going to update web tests...")
     ## location where test_*.html files are:
-    testdir = conf.get("Invenio", 'CFG_PREFIX') + os.sep + \
+    testdir = conf.get("Invenio", '_CFG_PREFIX') + os.sep + \
```
Comment (jirikuncar) [inline invenio/legacy/inveniocfg.py]: no change here -- should be deprecated anyway

File: invenio/legacy/websession/inveniogc.py [hunk 4]
```diff
@@ -32,7 +32,7 @@
 
 from invenio.legacy.dbquery import run_sql, wash_table_column_name
 from invenio.config import CFG_LOGDIR, CFG_TMPDIR, CFG_CACHEDIR, \
-        CFG_TMPSHAREDDIR, CFG_WEBSEARCH_RSS_TTL, CFG_PREFIX, \
+        CFG_TMPSHAREDDIR, CFG_WEBSEARCH_RSS_TTL, _CFG_PREFIX, \
```
Comment (jirikuncar) [inline invenio/legacy/websession/inveniogc.py]: :recycle:

**Comment by jirikuncar** [comment]
@dset0x it looks good. Please remove WIP and fix kwalitee except in `invenio/legacy/*` files.

**Comment by jirikuncar** [comment]
@dset0x it looks fine. Can you polish commit messages (^(bib|web).* -> legacy, testsuite -> tests)?

**Comment by jirikuncar** [comment]
ping @dset0x 
(cc @jalavik @jmartinm)

**Comment by jalavik** [comment]
So there is a few different things here, but it looks all fine for me. However I did not yet try the new `demosite populate` or other refactored commands - so will need to do that.

**Comment by jirikuncar** [comment]
There are still some problems with tests. Can you check this one?
```
======================================================================
ERROR: test_usage (invenio.testsuite.test_legacy_docextract_convert_journals.ScriptTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/inveniosoftware/invenio/invenio/testsuite/test_legacy_docextract_convert_journals.py", line 61, in test_usage
    out = run_py_func(["convert_journals", "-h"], main).out
  File "/home/travis/build/inveniosoftware/invenio/invenio/base/utils.py", line 307, in run_py_func
    manager_run()
TypeError: 'list' object is not callable

----------------------------------------------------------------------
```
The rest should be solved after you rebase.

**Comment by jirikuncar** [comment]
@dset0x please improve commit messages (`kwalitee check message master..upstream/pr/2829`).

- base -> global
- (bib|web)* -> legacy

**Comment by jirikuncar** [comment]
@dset0x can you rebase and amend the component names? Thanks

File: invenio/base/config.py [hunk 5]
```diff
@@ -77,26 +80,39 @@
     'invenio.legacy.websubmit',
 ]
 
-CFG_PREFIX = distutils.sysconfig.get_config_var("prefix")
+_cfg_prefix = distutils.sysconfig.get_config_var("prefix")
+
+
+@deprecated(
+    'Use a more specific variable from invenio/base/config.py instead',
+    RemovedInInvenio22Warning)
```
Comment (jirikuncar) [inline invenio/base/config.py]: I would just change it to `RemovedInInvenio23Warning`

File: invenio/legacy/batchuploader/cli.py [hunk 6]
```diff
@@ -67,12 +66,10 @@ def task_run_core():
         and uploads them.
         Files are then moved to the corresponding DONE folders.
     """
-    daemon_dir = CFG_BATCHUPLOADER_DAEMON_DIR[0] == '/' and CFG_BATCHUPLOADER_DAEMON_DIR \
-                 or CFG_PREFIX + '/' + CFG_BATCHUPLOADER_DAEMON_DIR
     # Check if directory /batchupload exists
     if not task_get_option('documents'):
         # Metadata upload
-        parent_dir = daemon_dir + "/metadata/"
+        parent_dir = os.path.join(CFG_BATCHUPLOADER_DAEMON_DIR, "metadata/")
```
Comment (jirikuncar) [inline invenio/legacy/batchuploader/cli.py]: maybe you want to fix also the trailing `/`

Comment (jirikuncar) [inline invenio/legacy/batchuploader/cli.py]: or just forget it as the legacy module will be soon removed

File: invenio/legacy/webcomment/webinterface.py [hunk 7]
```diff
@@ -857,15 +856,16 @@ def _get(self, req, form):
...
+                                                str(argd['comid']),
+                                                argd['file']))
 
             # Check that we are really accessing attachements
             # directory, for the declared record.
-            if path.startswith(CFG_PREFIX + '/var/data/comments/' + \
-                               str(self.recid)) and \
-                   os.path.exists(path):
+            if path.startswith(os.path.join(CFG_COMMENTSDIR,
+                               str(self.recid))) and \
```
Comment (jirikuncar) [inline invenio/legacy/webcomment/webinterface.py]: the indentation is weird here ...

File: invenio/legacy/webdoc/api.py [hunk 8]
```diff
@@ -739,7 +739,8 @@ def get_mo_last_modification():
     """
     # Take one of the mo files. They are all installed at the same
     # time, so last modication date should be the same
-    mo_file = '%s/share/locale/%s/LC_MESSAGES/invenio.mo' % (CFG_PREFIX, CFG_SITE_LANG)
+    mo_file = os.path.join(CFG_LOCALEDIR,
```
Comment (jirikuncar) [inline invenio/legacy/webdoc/api.py]: `os.path.join(CFG_LOCALEDIR, CFG_SITE_LANG, 'LC_MESSAGES', 'invenio.mo')`

File: invenio/legacy/websubmit/webinterface.py [hunk 9]
```diff
@@ -560,9 +560,9 @@ def attachfile(self, req, form):
                            'filetype': filetype}
 
         # Path to directory where uploaded files are saved
-        user_files_absolute_path = '%(CFG_PREFIX)s/var/tmp/attachfile/%(uid)s/%(filetype)s' % \
+        user_files_absolute_path = '%(CFG_TMPDIR)s/attachfile/%(uid)s/%(filetype)s' % \
```
Comment (jirikuncar) [inline invenio/legacy/websubmit/webinterface.py]: os.path.join(CFG_TMPDIR, 'attachfile', uid, filetype)

File: invenio/legacy/websubmit/webinterface.py [hunk 10]
```diff
@@ -666,13 +666,13 @@ def getattachedfile(self, req, form):
...
-                                   '/'  + str(argd['uid']) + \
-                                   '/' + argd['type'] + '/' + argd['file'])
+            path = os.path.abspath(os.path.join(CFG_TMPDIR, 'attachfile',
+                                                str(argd['uid']),
+                                                argd['type'], argd['file']))
 
             # Check that we are really accessing attachements
             # directory, for the declared record.
-            if path.startswith(CFG_PREFIX + '/var/tmp/attachfile/') and os.path.exists(path):
+            if path.startswith(os.path.join(CFG_TMPDIR, 'attachfile/')) and os.path.exists(path):
```
Comment (jirikuncar) [inline invenio/legacy/websubmit/webinterface.py]: trailing `/`?

**Pull request merged**